src/simulated_bifurcation/polynomial/expression_compiler.py

```diff
@@ -0,0 +1,104 @@
+from typing import Tuple
+
+import numpy as np
+import torch
+from sympy import Poly
+
+
+class ExpressionCompiler:
+    """
+    Helper class to convert Sympy quadratic multivariate polynomial
+    expressions to tensors that respectively represent the constant,
+    linear and quadratic coefficients of said polynomial.
+
+    The tensors generated are understandable by SB's native polynomials
+    objects.
+
+    The aim of this class is to define polynomials in a more natural way,
+    using mathematical equations that are more accessible and less
+    cumbersome than tensors, especially for small instances.
+
+    Notes
+    -----
+    This class only handles order 2 polynomials.
+
+    See Also
+    --------
+    BaseMultivariateQuadraticPolynomial
+    """
+
+    def __init__(self, polynomial: Poly) -> None:
+        """
+        Parameters
+        ----------
+        expression : Sympy
+            the natural mathematical writing of the polynomial.
+        """
+        ExpressionCompiler.__check_polynomial_degree(polynomial)
+        self.polynomial = polynomial
+        self.variables = len(self.polynomial.gens)
+        self.__quadratic_coefficients = torch.zeros(self.variables, self.variables)
+        self.__linear_coefficients = torch.zeros(self.variables)
```

Comment:
```suggestion
        self.__linear_coefficients = torch.zeros(self.n_variables)
```
see above suggestion